src/hooks/useAddCasesInputs.tsx

```diff
@@ -0,0 +1,141 @@
+import { startOfDay } from "date-fns";
+import { pick } from "lodash";
+import { useCallback, useEffect, useState } from "react";
+
+import { saveFacility } from "../database";
+import {
+  ModelInputsPopulationBrackets,
+  populationBracketKeys,
+} from "../impact-dashboard/EpidemicModelContext";
+import useModel from "../impact-dashboard/useModel";
+import { Facility, ModelInputs } from "../page-multi-facility/types";
+import useFacilityModelVersions from "./useFacilityModelVersions";
+
+export const getModelInputs = (modelInputs: ModelInputs) => {
+  return pick(modelInputs, populationBracketKeys);
+};
+
+const useAddCasesInputs = (
+  facility: Facility,
+  onSave: (f: Facility) => void,
+  observedAt?: Date | undefined,
+) => {
+  const [facilityModelVersions, updateModelVersions] = useFacilityModelVersions(
+    facility,
+  );
+  const [observedAtVersion, setObservedAtVersion] = useState<
+    ModelInputs | undefined
+  >();
+  // the current state of the facility is the default when we need to reset
+  // If observedAt is provided and a version exists then use the version's inputs
+  const defaultInputs = getModelInputs(
+    observedAtVersion || facility.modelInputs,
+  );
+  // If observedAt is provided then set it as the default date
+  const defaultObservationDate = observedAt || facility.modelInputs.observedAt;
+
+  const [inputs, setInputs] = useState<ModelInputsPopulationBrackets>(
+    defaultInputs,
+  );
+
+  const [observationDate, setObservationDate] = useState(
+    defaultObservationDate,
+  );
+
+  // maybe export this so that we don't define it twice (it is also in the AddCasesModalContent component)
```

Comment:
Would be good, but I also don't mind filing a tech debt issue to loop back around and handle this, given how much effort has already gone into this feature and its overall importance!

Reply:
Okay sounds good, I'll remove the note for now.